From a9aa235a1034d88e8febab169b7a9a7dd33479fd Mon Sep 17 00:00:00 2001
From: Francois-Xavier Coudert <fxcoudert@gcc.gnu.org>
Date: Mon, 7 Sep 2020 09:38:25 +0200
Subject: [PATCH] fortran: Fix argument types in derived types procedures

gcc/fortran/ChangeLog

	* trans-types.c (gfc_get_derived_type): Fix argument types.

(cherry picked from commit d30869a8d4886aee4020be3b28b15b1b15c8d9ad)
---
 gcc/fortran/trans-types.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git gcc/fortran/trans-types.c gcc/fortran/trans-types.c
index 7bf28f49588..8248d67064e 100644
--- gcc/fortran/trans-types.c
+++ gcc/fortran/trans-types.c
@@ -2712,7 +2712,7 @@ gfc_get_derived_type (gfc_symbol * derived, int codimen)
 	field_type = gfc_get_ppc_type (c);
       else if (c->attr.proc_pointer && derived->backend_decl)
 	{
-	  tmp = build_function_type_list (derived->backend_decl, NULL_TREE);
+	  tmp = build_function_type (derived->backend_decl, NULL_TREE);
 	  field_type = build_pointer_type (tmp);
 	}
       else if (c->ts.type == BT_DERIVED || c->ts.type == BT_CLASS)
-- 
2.40.1

